components/camera/transformpipeline/mods.go

```diff
@@ -55,13 +66,15 @@ func (rs *rotateSource) Read(ctx context.Context) (image.Image, func(), error) {
 	}
 	switch rs.stream {
 	case camera.ColorStream, camera.UnspecifiedStream:
-		return imaging.Rotate(orig, 180, color.Black), release, nil
+		// imaging.Rotate rotates an image counter-clockwise but our rotate function rotates in the
+		// clockwise direction. The angle is negated here for consistency.
+		return imaging.Rotate(orig, -(rs.angle), color.Black), release, nil
```

Comment:
Should there be a check for rotations that are not 90 & 270 for color images? The imaging package can do rotations that are any number, but I'm wondering if we should limit it to just be the purely 90 degree rotations

Reply:
I don't think so. There isn't a good reason to limit the functionality that we already support. We _could_ error if users try to rotate color images 45 degrees or 22.5 but we could also just do it and only error if there's actually a reason we can't like the new width x height can't be used with the x264 codec.